src/middleware/rpi_fwupdate.rs

```diff
@@ -0,0 +1,177 @@
+use std::{path::PathBuf, time::Duration};
+use tokio::{
+    fs::File,
+    io::{AsyncRead, AsyncWrite},
+    sync::mpsc::Sender,
+    time::sleep,
+};
+
+use super::{
+    firmware_update_usb::FwUpdate,
+    usbboot::{FlashProgress, FlashStatus, FlashingError},
+};
+
+pub struct RpiFwUpdate {
+    msd_device: File,
+}
+
+impl RpiFwUpdate {
+    pub const VID_PID: (u16, u16) = (0x2207, 0x350b);
```

Comment:
These are copy-pasted from RK1's module

Reply:
these are the correct values for rockchip USB gadget.